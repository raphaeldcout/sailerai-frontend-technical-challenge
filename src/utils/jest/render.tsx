import { FC, ReactElement, ReactNode } from 'react';

import { ThemeProviderSailer } from '@/styles/withThemeProvider';
import { render, RenderOptions } from '@testing-library/react';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <ThemeProviderSailer>{children}</ThemeProviderSailer>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };

