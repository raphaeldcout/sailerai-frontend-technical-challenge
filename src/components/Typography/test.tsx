import '@testing-library/jest-dom';

import { render } from '@/utils/jest/render';

import { Typography } from '.';

describe('Typography', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <Typography>Lorem ipsum is typically.</Typography>
    );

    expect(container).toMatchSnapshot();
  });
});
