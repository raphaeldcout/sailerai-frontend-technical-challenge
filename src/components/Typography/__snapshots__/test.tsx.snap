// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should match snapshot 1`] = `
<div>
  <p
    class="sc-dc096805-0 lbvuzb"
    color="surface"
    variant="medium"
    weight="regular"
  >
    Lorem ipsum is typically.
  </p>
</div>
`;
